### PR TITLE
Add functions.c

### PR DESCRIPTION
**Pull request by MCJOHN974**

### CODE REVIEW

File: objects.cpp [hunk 1]
```diff
@@ -68,8 +71,8 @@ class FullCart : public Cart {
     }
     Cart* recalc(int country) override {
         return new FullCart(
-            this->before->recalc(country),
-            this->item->prepare(country)
+                this->before->recalc(country),
+                this->item->prepare(country)
```
Comment (yegor256) [inline objects.cpp:79]: @MCJOHN974 why? four spaces everywhere

File: objects.cpp [hunk 2]
```diff
@@ -29,6 +29,9 @@ class Digital : public Stocked {
     int deliver() override {
         return Stocked::deliver() / 2;
     }
+    Item* prepare(int country) override {
+        return new Digital(stk);
```
Comment (yegor256) [inline objects.cpp]: @MCJOHN974 this is wrong. You should use "copy" constructor, like it works in `Tangible`

File: objects.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ class Stocked : public Item {
         this->stk->total--;
         return this->stk->price;
     }
-private:
+protected:
```
Comment (yegor256) [inline objects.cpp]: @MCJOHN974 bad idea always. Never use `protected` in OOP

**Comment by yegor256** [comment]
@MCJOHN974 a few comments above. Overall, you don't use recursion to its full potential. I suggest getting rid of `int*`

**Comment by MCJOHN974** [comment]
I remove ```protected```, replace tabs to spaces, and now both ```functions``` and ```objects``` can be build by make

File: functions.c [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    if (begin == end) {
+        return 0;
+    } else {
+        return DeliverDigitals(begin + 1, end, discounts) + stocks[begin].price / 2;
+    }
+}
+
+int DeliverTangibles(int begin, int end, int* discounts) {
+    PrepareTangibles(begin, end, discounts);
+
```
Comment (yegor256) [inline functions.c]: @MCJOHN974 see https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

File: functions.c [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#include "stocks.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+
+int Digitals(int begin, int end, int country, int* discounts) {
+    if (begin == end) {
+        return 0;
```
Comment (yegor256) [inline functions.c:8]: @MCJOHN974 see https://www.yegor256.com/2015/01/21/if-then-throw-else.html

File: functions.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (begin == end) {
+        return 0;
+    } else {
+        return DeliverDigitals(begin + 1, end, discounts) + stocks[begin].price / 2;
+    }
+}
+
+int DeliverTangibles(int begin, int end, int* discounts) {
+    PrepareTangibles(begin, end, discounts);
```
Comment (yegor256) [inline functions.c:42]: @MCJOHN974 this is procedural style right here

**Comment by yegor256** [comment]
@MCJOHN974 see my comments above

**Comment by MCJOHN974** [comment]
@yegor256 I add counting of left items in objects and functions and fix bad codestyle

**Pull request merged**